force-app/main/default/classes/CaseRoutingService.cls

```diff
@@ -0,0 +1,80 @@
+public with sharing class CaseRoutingService {
+    
+    private static final Map<LanguageEnum, SObjectField> languageToQueueMap = new Map<LanguageEnum, SObjectField>{
+        LanguageEnum.English => Routing_Configuration__mdt.Routing_English__c,
+        LanguageEnum.Chinese => Routing_Configuration__mdt.Routing_Chinese__c,
+        LanguageEnum.German => Routing_Configuration__mdt.Routing_German__c,
+        LanguageEnum.French => Routing_Configuration__mdt.Routing_French__c,
+        LanguageEnum.Italian => Routing_Configuration__mdt.Routing_Italian__c,
+        LanguageEnum.Portuguese => Routing_Configuration__mdt.Routing_Portuguese__c
+    };
+
+    private static CaseRoutingService instance;
+    private static Map<String, Group> queuesByDeveloperName;
+    private static Map<Id, Group> queues;
+
+public static CaseRoutingService getInstance() {
+
+    if (CaseRoutingService.instance == null) {
+        CaseRoutingService.instance = new CaseRoutingService();
+    }
+    return CaseRoutingService.instance;
+}
+
+public Id getQueueForCase(Case newCase) {
+    if (!queues.containsKey(newCase.OwnerId)) {
```

Comment:
Maybe it would be better to actually check if `queues` does contain the Id? That way you are saving usage of return statements and code is shorter.
It would also be easier to provide code coverage with tests this way.